### PR TITLE
[kevent 4/5] Implement select syscall as a wrapper of kevent

### PR DESCRIPTION
**Pull request by xThaid**
As it turned out, [select](https://man.netbsd.org/NetBSD-6.0/select.2) system call can be implemented as a libc wrapper for kevent. Maybe it isn't the most efficient solution, but will do.

### CODE REVIEW

File: lib/libc/sys/select.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    int nevents = 0;
+
+    if (timeout != NULL)
+        tv2ts(timeout, &timeout_ts);
+
+    kq = kqueue();
+    if (kq < 0)
+        return -1;
+
+    events = malloc(2 * nfds * sizeof(struct kevent));
```
Comment (wiklam) [inline lib/libc/sys/select.c:23]: Handle situation when malloc return NULL.
You have to check if nfds is a correct argument ([EINVAL](https://man7.org/linux/man-pages/man2/select.2.html))

File: lib/libc/sys/select.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+int select(int nfds, fd_set *restrict readfds, fd_set *restrict writefds,
+           fd_set *restrict exceptfds, struct timeval *restrict timeout) {
+    int kq;
+    int ret;
+    struct timespec timeout_ts;
+    struct kevent *events;
+    int nevents = 0;
+
+    if (timeout != NULL)
```
Comment (wiklam) [inline lib/libc/sys/select.c:16]: Check timeout too.

**Review by wiklam — CHANGES_REQUESTED** [review]
What about incompatible kevent/kqueue errors?

File: lib/libc/sys/select.c (hunk 2 above)
Comment (xThaid) [reply lib/libc/sys/select.c:21]: Could you elaborate on that? What should I check? The specification of `select` syscall allows all values of `timeout`.

Comment (wiklam) [reply lib/libc/sys/select.c:21]: I was thinking about returning `EINVAL` for negative and large values of timeout.

Comment (xThaid) [reply lib/libc/sys/select.c:21]: I see. The specification however, doesn't say anything about big nor negative values.

Comment (cahirwpz) [reply lib/libc/sys/select.c:21]: NetBSD's [select](https://man.netbsd.org/select.2) manual states:

> [EINVAL] The specified time limit is invalid.  One of its components is negative or too large.

Comment (xThaid) [reply lib/libc/sys/select.c]: This is interesting. I have no idea where I found that information a few months ago.

**Review by cahirwpz — APPROVED** [review]

**Pull request merged**